unit/src/MooseServerTest.C

```diff
@@ -0,0 +1,531 @@
+                          std::stringstream & diagnostics_stream) const
+  {
+    size_t diagnostics_size = diagnostics_array.size();
+
+    for (size_t i = 0; i < diagnostics_size; i++)
+    {
+      std::stringstream diagnostic_errors;
+      int diagnostic_start_line;
+      int diagnostic_start_character;
+      int diagnostic_end_line;
+      int diagnostic_end_character;
+      int diagnostic_severity;
+      std::string diagnostic_code;
+      std::string diagnostic_source;
+      std::string diagnostic_message;
+
+      ASSERT_TRUE(wasp::lsp::dissectDiagnosticObject(*(diagnostics_array.at(i).to_object()),
+                                                     diagnostic_errors,
+                                                     diagnostic_start_line,
+                                                     diagnostic_start_character,
+                                                     diagnostic_end_line,
+                                                     diagnostic_end_character,
+                                                     diagnostic_severity,
+                                                     diagnostic_code,
+                                                     diagnostic_source,
+                                                     diagnostic_message));
+
+      diagnostics_stream << "line:" << diagnostic_start_line
+                         << " column:" << diagnostic_start_character << " - " << diagnostic_message
+                         << std::endl;
+    }
+  }
+
+  // traverse symbolpaths and make formatted list for easy testing and viewing
+  void format_symbolpaths(const wasp::DataObject & symbols_response,
+                          std::stringstream & paths_stream) const
+  {
+    ASSERT_TRUE(wasp::lsp::verifySymbolsResponse(symbols_response));
+    wasp::lsp::SymbolIterator si(std::make_shared<wasp::DataObject>(symbols_response));
+
+    for (std::vector<int> indices{0}; indices.back() < (int)si.getChildSize(); indices.back()++)
+    {
+      si.moveToChildAt(indices.back());
+      indices.push_back(-1);
+      paths_stream << si.getPath() << std::endl;
+      if (si.getChildSize() == 0)
+      {
+        while (indices.back() + 1 == (int)si.getChildSize() && si.moveToParent())
+        {
+          indices.pop_back();
+        }
+      }
+    }
+  }
+
+  // create moose_unit_app and moose_server to persist for reuse between tests
+  static void SetUpTestCase()
+  {
+    moose_unit_app = AppFactory::createAppShared("MooseUnitApp", 0, nullptr);
+    moose_server = std::make_unique<MooseServer>(*moose_unit_app);
+  }
+
+  // delete moose_unit_app and moose_server after all test runs have completed
+  static void TearDownTestCase()
+  {
+    moose_unit_app.reset();
+    moose_server.reset();
+  }
+
+  // statically declare moose_unit_app and moose_server for reuse across tests
+  static std::shared_ptr<MooseApp> moose_unit_app;
+  static std::unique_ptr<MooseServer> moose_server;
```

Comment:
Is static the gest-approved way to do this? It doesn't seem thread safe, but... I guess this isn't threaded

Reply:
Yes, the gtest `void SetUpTestCase()` and `void TearDownTestCase()` have to be defined as static methods.

And these static methods are only allowed to operate on static member variable data.

So static is not only the recommended way to do data setup to be reused across all test cases, it is required.